src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.h

```diff
@@ -600,6 +600,42 @@ class CameraAVStreamMgmtServer : public CommandHandlerInterface, public Attribut
         return CHIP_NO_ERROR;
     }
 
+    template <typename StreamContainer, typename IdGetter>
+    bool ValidateStreamForModifyOrDeallocateImpl(StreamContainer & streams, uint16_t streamID, HandlerContext & ctx,
+                                                 const char * streamTypeName, IdGetter id_getter)
+    {
+        auto it = std::find_if(streams.begin(), streams.end(), [&](const auto & stream) { return id_getter(stream) == streamID; });
+
+        if (it == streams.end())
+        {
+            ChipLogError(Zcl, "CameraAVStreamMgmt[ep=%d]: %s stream with ID: %u not found", mEndpointId, streamTypeName, streamID);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::NotFound);
+            return false;
+        }
+
+        if (it->referenceCount > 0)
+        {
+            ChipLogError(Zcl, "CameraAVStreamMgmt[ep=%d]: %s stream with ID: %u still in use", mEndpointId, streamTypeName,
+                         streamID);
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::InvalidInState);
+            return false;
+        }
+
+        using StreamType = typename StreamContainer::value_type;
+        if constexpr (std::is_same_v<StreamType, VideoStreamStruct> || std::is_same_v<StreamType, AudioStreamStruct>)
+        {
+            if (it->streamUsage == Globals::StreamUsageEnum::kInternal)
```

Comment:
Where is this check in the spec for Modify?  I don't see it there.

Or at least it's equally there or not for both snapshot and audio/video streams (unlike Deallocate, where it's there for audio/video but not snapshot).

Or is the point that snapshot streams are never internal?  I see nothing in the spec that guarantees that.

Reply:
Apologies. There is a bit of flux here. It is there in the current Spec for Audio/Video StreamDeallocate. And it is getting added for `VideoStreamModify` in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/12117. Expected the Spec PR to have/be merged soon.
SnapshotStreams, OTOH, can only be Internal if they are being derived from a VideoStream that is Internal. In that scenario, SnapshotStreamModify would be disallowed. This bug was found recently and filed here: https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/12126.
Let's hold off on the review until the Spec side is in then as I would need to update the SnapshotStreamModify as well.

Reply:
Ah, the usual "we are implementing an in-flight PR but not linking to it" thing?  Alright then.